src/build/_exceptions.py

```diff
@@ -64,6 +64,30 @@ def __str__(self) -> str:
         return description
 
 
+class CircularBuildSystemDependencyError(BuildException):
```

Comment:
This is any build dependency, not just build system dependencies.

```suggestion
class CircularBuildDependencyError(BuildException):
```